lib/chains/llm_chain.ex

```diff
@@ -317,6 +317,11 @@ defmodule LangChain.Chains.LLMChain do
   """
   @spec add_message(t(), Message.t()) :: t()
   def add_message(%LLMChain{} = chain, %Message{} = new_message) do
+    if Message.has_images?(new_message) and chain.llm.model != "gpt-4-vision-preview" do
```

Comment:
We don't want model-specific details in LLMChain. We want LLMChain to be model agnostic.